### DISCUSSION TITLE
Does the expression evaluate date fields if one greater than the other?

### DISCUSSION THREAD

**Discussion by malola** [microsoft/RulesEngine#135] [Q&A]
Does the expression allow including date values and does it evaluate if one date greater than the other?
If i were to check if one date is greater than the other by x number of days, does expression allow to type cast string to date?

**Comment by abbasc52** [comment]
you can do `DateTime.Parse(date1)` to convert it datetime object and them you can use all of datetime methods for like `AddDays` etc.

To make it cleaner , you can use scopedparams to make it look cleaner

```c#
{
      "RuleName": "compare dates",
      "LocalParams":[
        {
          "Name":"date1",
          "Expression":"DateTime.Parse(input1)"
        },{
            "Name":"date2",
            "Expression":"DateTime.Parse(input2)"
       }
      ],
      "Expression":"date1.AddDays(x) > date2"
    }
```
